dragonfly/building.py

```diff
@@ -296,6 +468,89 @@ def to_dict(self, abridged=False, included_prop=None):
         base['properties'] = self.properties.to_dict(abridged, included_prop)
         return base
 
+    @staticmethod
+    def _generate_room_2ds(face3d_array, flr_to_flr, bldg_name, flr_count, tolerance):
+        """Generate Room2D objects given geometry and information about their parent."""
+        room_2ds = []
+        for i, room_geo in enumerate(face3d_array):
+            room = Room2D('{}_Floor{}_Room{}'.format(bldg_name, flr_count, i + 1),
+                          room_geo, flr_to_flr, tolerance=tolerance)
+            room_2ds.append(room)
+        return room_2ds
+
+    @staticmethod
+    def _is_story_equivalent(face1, face2, tolerance):
+        """Check whether the area, XY centerpoint and XY first point match between Face3D.
```

Comment:
Isn't it more straight forward to just check all the vertices?

Reply:
I have [some methods in ladybug_geometry that test for complete geometric equivalency](https://www.ladybug.tools/ladybug-geometry/docs/ladybug_geometry.geometry3d.face.html#ladybug_geometry.geometry3d.face.Face3D.is_geometrically_equivalent) but they are overkill for all of the situations that I have encountered thus far and they really don't scale very well to urban-sized models where you have lots of repeated floors. If this method ends up producing some situations that are misaligned with people's expectations we can add an option to the classmethods to use these more detailed methods for matching vertices. But, at this point, this method is preferable in the workflows that I have tried.